test/integration/load/load_test.go

```diff
@@ -60,142 +63,138 @@ todo: consider adding the following scenarios
+	}
 }
 
 // TestValidateState validates the state file based on the os and cni type.
 func TestValidateState(t *testing.T) {
 	clientset, err := kubernetes.MustGetClientset()
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
+
 	config := kubernetes.MustGetRestConfig(t)
 	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Minute)
 	defer cancel()
 
-	validator, err := validate.CreateValidator(ctx, clientset, config, namespace, *cniType, *restartCase, *osType)
-	if err != nil {
-		t.Fatal(err)
-	}
-	if err := validator.Validate(ctx); err != nil {
-		t.Fatal(err)
+	validator, err := validate.CreateValidator(ctx, clientset, config, namespace, testConfig.CNIType, testConfig.RestartCase, testConfig.OSType)
+	require.NoError(t, err)
+
+	err = validator.Validate(ctx)
+	require.NoError(t, err)
+
+	if testConfig.Cleanup {
+		err = validator.Cleanup(ctx)
+		require.NoError(t, err, "failed to cleanup validator")
 	}
 }
 
 // TestScaleDeployment scales the deployment up/down based on the replicas passed.
-// go test -timeout 30m -tags load -run ^TestScaleDeployment$ -tags=load -replicas 10
+// REPLICAS=10 go test -timeout 30m -tags load -run ^TestScaleDeployment$ -tags=load
 func TestScaleDeployment(t *testing.T) {
 	t.Log("Scale deployment")
 	clientset, err := kubernetes.MustGetClientset()
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
+
 	ctx := context.Background()
 	// Create namespace if it doesn't exist
 	namespaceExists, err := kubernetes.NamespaceExists(ctx, clientset, namespace)
-	if err != nil {
-		t.Fatal(err)
-	}
+	require.NoError(t, err)
 
 	if !namespaceExists {
 		err = kubernetes.MustCreateNamespace(ctx, clientset, namespace)
-		if err != nil {
-			t.Fatal(err)
-		}
+		require.NoError(t, err)
 	}
 
-	deployment, err := kubernetes.MustParseDeployment(noopDeploymentMap[*osType])
-	if err != nil {
-		t.Fatal(err)
+	deployment, err := kubernetes.MustParseDeployment(noopDeploymentMap[testConfig.OSType])
+	require.NoError(t, err)
+
+	if testConfig.Cleanup {
+		deploymentsClient := clientset.AppsV1().Deployments(namespace)
+		err = kubernetes.MustCreateDeployment(ctx, deploymentsClient, deployment)
+		require.NoError(t, err)
 	}
+
 	deploymentsClient := clientset.AppsV1().Deployments(namespace)
-	err = kubernetes.MustScaleDeployment(ctx, deploymentsClient, deployment, clientset, namespace, podLabelSelector, *replicas, *skipWait)
-	if err != nil {
-		t.Fatal(err)
+	err = kubernetes.MustScaleDeployment(ctx, deploymentsClient, deployment, clientset, namespace, podLabelSelector, testConfig.Replicas, testConfig.SkipWait)
+	require.NoError(t, err)
+
+	if testConfig.Cleanup {
+		err = kubernetes.MustDeleteDeployment(ctx, deploymentsClient, deployment)
+		require.NoError(t, err, "error deleteing load deployment")
 	}
 }
 
 func TestDualStackProperties(t *testing.T) {
-	clientset, err := kubernetes.MustGetClientset()
-	if err != nil {
-		t.Fatal(err)
+	if !testConfig.ValidateDualStack {
```

Comment:
Why do we need this ?

Reply:
Ah, because now, with that `make test-load` target, it runs all tests in `load_test.go`
If we don't have this check, it will try to validate dualstack stuff, every time, but that's not the case we want, we only want this test to run if the config is set as a dualstack cluster